include/klee/util/ArrayCache.h

```diff
@@ -0,0 +1,89 @@
+//===-- ArrayCache.h --------------------------------------------*- C++ -*-===//
+//
+//                     The KLEE Symbolic Virtual Machine
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef KLEE_ARRAY_CACHE_H
+#define KLEE_ARRAY_CACHE_H
+
+#include "klee/Expr.h"
+
+// FIXME: Remove this hack when we switch to C++11
+#ifdef _LIBCPP_VERSION
+#include <unordered_set>
+#define unordered_set std::unordered_set
+#else
+#include <tr1/unordered_set>
+#define unordered_set std::tr1::unordered_set
+#endif
+
+#include <string>
+#include <vector>
+
+namespace klee {
+
+// FIXME: This is copied from ``klee/util/ArrayExprHash.h``.
```

Comment:
@MartinNowack 
Given that the switch to C++11 is blocked (#314). I take it you would be okay with the hack above for now? You will see something very similar in ``include/klee/util/ArrayExprHash.h``.